cpp/src/Ice/InputStream.cpp

```diff
@@ -2677,11 +2673,10 @@ Ice::InputStream::EncapsDecoder11::readInstance(int32_t index, PatchFunc patchFu
         //
         if (!_sliceValues)
         {
-            throw NoValueFactoryException(
+            throw MarshalException{
                 __FILE__,
                 __LINE__,
-                "no value factory found and value slicing is disabled",
```

Comment:
I don't know why we give this option - I find this very confusing.

For me, slicing or not slicing should be only up to the encoder. The decoder should not have a say.